tests/test_spam_admin.py

```diff
@@ -0,0 +1,440 @@
+
+
+    @httpretty.activate
+    def test_comment_added_is_spam(self):
+        comment = self._add_comment(
+            self.project, content = self.GTUBE, auth=self.project.creator.auth,
+        )
+        self.project.reload()
+
+
+
+        assert_equal(len(self.project.commented), 1)
+
+        if self.spam_assassin_active:
+            assert_equal(comment.spam_status, Comment.POSSIBLE_SPAM)
+        else:
+            assert_equal(comment.spam_status, Comment.UNKNOWN)
+
+    @httpretty.activate
+    def test_comment_added_is_not_spam(self):
+        comment = self._add_comment(
+            self.project,auth=self.project.creator.auth,
+        )
+        self.project.reload()
+
+
+
+        assert_equal(len(self.project.commented), 1)
+        assert_equal(comment.spam_status , Comment.UNKNOWN)
+
+    @httpretty.activate
+    def test_train_spam_comment(self):
+
+        httpretty.register_uri(
+            httpretty.POST, SPAM_ASSASSIN_TEACHING_URL,
+            body=lambda r, u, h: (200, h,json.dumps({"status":"Learned"}))
+        )
+
+        comment = self._add_comment(
+            self.project,
+            content=self.GTUBE,
+            auth=self.project.creator.auth,
+        )
+        self.project.reload()
+
+
+        if self.spam_assassin_active:
+            assert_true(train_spam(comment, is_spam=True))
+        else:
+            assert_false(train_spam(comment, is_spam=True))
+
+
+    @httpretty.activate
+    def test_train_ham_comment(self):
+
+        httpretty.register_uri(
+            httpretty.POST, SPAM_ASSASSIN_TEACHING_URL,
+            body=lambda r,u,h: (200,h,json.dumps({"status":"Learned"}))
+        )
+        comment = self._add_comment(
+            self.project, auth=self.project.creator.auth,
+        )
+        if self.spam_assassin_active:
+            assert_true(train_spam(comment, is_spam=False))
+        else:
+            assert_false(train_spam(comment, is_spam=False))
+
+    @httpretty.activate
+    def test_auto_mark_spam_if_flagged_enough_times(self):
+        comment = self._add_comment(
+            self.project, auth=self.project.creator.auth,
+        )
+
+
+        for i in range(Comment.NUM_FLAGS_FOR_SPAM):
+            reporter = AuthUserFactory()
+            url = self.project.api_url + 'comment/{0}/report/'.format(comment._id)
+            self.app.post_json(
+                url,
+                {
+                    'category': 'spam',
+                    'text': 'ads',
+                },
+                auth=reporter.auth,
+            )
+
+        comment.reload()
+        assert_equal(comment.spam_status , Comment.POSSIBLE_SPAM)
+
+    @httpretty.activate
+    def test_dont_auto_mark_spam_if_not_flagged_enough_times(self):
+        comment = self._add_comment(
+            self.project, auth=self.project.creator.auth,
+        )
+
+
+        for i in range(Comment.NUM_FLAGS_FOR_SPAM-1):
```

Comment:
See previous comment. Where 1 is an edge-case, 0 means that this code won't even run once meaning that the post won't happen and the code that it is testing won't be run a single time. Using a larger number for test cases will ensure that the code is run and it won't trip something in the first run.